evm/ownership.js

```diff
@@ -0,0 +1,200 @@
+            }
+
+            try {
+                await ownershipContract.transferOwnership(newOwner).then((tx) => tx.wait());
+            } catch (error) {
+                throw new Error(error);
+            }
+
+            owner = await ownershipContract.owner();
+
+            if (owner.toLowerCase() !== newOwner.toLowerCase()) {
+                throw new Error('Ownership transfer failed.');
+            } else {
+                console.log(`New contract owner: ${owner}`);
+            }
+
+            break;
+        }
+
+        case 'proposeOwnership': {
+            const owner = await ownershipContract.owner();
+
+            if (owner.toLowerCase() !== wallet.address.toLowerCase()) {
+                throw new Error(`Caller ${wallet.address} is not the contract owner.`);
+            }
+
+            if (!isAddress(newOwner) || newOwner === AddressZero) {
+                throw new Error(`Invalid new owner address: ${newOwner}`);
+            }
+
+            try {
+                await ownershipContract.proposeOwnership(newOwner).then((tx) => tx.wait());
+            } catch (error) {
+                throw new Error(error);
+            }
+
+            const pendingOwner = await ownershipContract.pendingOwner();
+
+            if (pendingOwner.toLowerCase() !== newOwner.toLowerCase()) {
+                throw new Error('Propose ownership failed.');
+            } else {
+                console.log(`New pending owner: ${pendingOwner}`);
+            }
+
+            break;
+        }
+
+        case 'acceptOwnership': {
+            const pendingOwner = await ownershipContract.pendingOwner();
+
+            if (pendingOwner === AddressZero) {
+                throw new Error('This is no pending owner.');
+            }
+
+            if (pendingOwner.toLowerCase() !== wallet.address.toLowerCase()) {
+                throw new Error(`Caller ${wallet.address} is not the pending owner.`);
+            }
+
+            try {
+                await ownershipContract.acceptOwnership().then((tx) => tx.wait());
+            } catch (error) {
+                throw new Error(error);
+            }
+
+            const newOwner = await ownershipContract.owner();
+
+            if (newOwner.toLowerCase() !== pendingOwner.toLowerCase()) {
+                throw new Error('Accept ownership failed.');
+            } else {
+                console.log(`New contract owner: ${newOwner}`);
+            }
+
+            break;
+        }
+
+        default: {
+            throw new Error(`Unknown ownership action ${ownershipAction}`);
+        }
+    }
+}
+
+async function main(options) {
+    const config = loadConfig(options.env);
+
+    const chain = options.chain;
+
+    if (config.chains[chain.toLowerCase()] === undefined) {
+        throw new Error(`Chain ${chain} is not defined in the info file`);
+    }
+
+    await processCommand(options, config.chains[chain.toLowerCase()], config);
```

Comment:
Function definition has 2 input args: `options`, `chain`. here we are supplying 3 args. We should remove `config` if not used.

Reply:
Good catch